src/binary/Builtin.cc

```diff
@@ -222,16 +222,37 @@ extern "C" {
   make_crypt_salt (const char* crypt_prefix, int crypt_rounds)
   {
     // use gensalt own auto entropy
-    const char *entropy = NULL;
-    const size_t entropy_len = 0;
+    char* retval = crypt_gensalt_ra (crypt_prefix, crypt_rounds, NULL, 0);
 
-    char* retval = crypt_gensalt_ra (crypt_prefix, crypt_rounds, entropy, entropy_len);
-
-    if (!retval)
+    // auto entropy might not be supported in some older systems (15.2 and older),
+    // if we get the "Invalid argument" error then read some entropy from
+    // /dev/urandom and try again
```

Comment:
maybe link this pr there? as there is no bugzilla, but it is good source of information

Reply:
I do not think it is needed, the comment above is enough IMHO. And the PR can be found from the GitHub history if needed...